packages/core/src/metrics/MetricsService.ts

```diff
@@ -65,22 +66,27 @@ export class MetricsService {
       throw new Error(`Type is invalid: ${type}`);
     }
 
+    if (!this.publisher || !this.defaultMetrics) {
+      return Promise.resolve();
```

Comment:
Don't we want to throw an error in this case?

Reply:
Also, why defaultMetrics are not always set by default? We can skip the check by calling `buildDefaultMetrics` always in the constructor.

Reply:
This is new requirements that are already implemented on android and IOS. Basically when config is not there there should be no errors. Thats classical scenario.

Reply:
The problem I see here is that you could be doing `metricsService.sendAppAndDevice()` and user would not get any feedback.

On the other hand, I think default metrics should be instantiated in constructor and never be undefined.

Reply:
If metrics is disabled then there is no need to collect or instantiate default metrics. No strong opinions about it. I just followed how I implemented that on IOS but there could be better way

Reply:
Ah.. get you now. As per recent req users should not interact with metrics. This metod is called on constructor

Reply:
Ah.. get you now. As per recent req users should not interact with metrics. This metod is called on constructor

Reply:
Ah.. get you now. As per recent req users should not interact with metrics. This metod is called on constructor

Reply:
We could a `console.warn("Metrics service disabled due to lack of configuration")` or something, no failures but a runtime warning.

Reply:
Yeah similar message is above this